### DISCUSSION TITLE
Apache Camel Karavan

### DISCUSSION THREAD

**Discussion by MN-Dheeraj** [apache/camel-karavan#1123] [Q&A] [answered]
I've started using Apache Camel Karavan. I am trying to build workflows(routes, logs etc) and run/export code. However, I have been seeing multiple issue.
The current issue I am facing is that I have written a small workflow integrating with my application.

PS D:\Apache-Camel-Karavan-Integration\projects\POC\Karavan_generated\karavan> jbang "-Dcamel.jbang.version=4.3.0" camel@apache/camel export --fresh  --directory=d:\Apache-Camel-Karavan-Integration\projects\POC\Karavan_generated\karavan\demosb

The runtime option must be specified

What is this runtime exactly? Can someone please help me?

**Comment by davsclaus** [comment]
Its whether its Spring Boot, Quarkus or Camel Main. Karavan uses main.

    --runtime=camel-main

**Comment by mgubaidullin** [comment]
Are using Karavan Vs Code extension or Web Application?
If VS Code, have you created an application?


<img width="397" alt="Screenshot 2024-02-14 at 10 57 00 AM" src="https://github.com/apache/camel-karavan/assets/1379213/cbcf5a77-c5ee-42e2-8727-36f79d2f9045">

**Reply by MN-Dheeraj** [reply]
Hello @mgubaidullin ,

Firstly, Thank you very much for your response.

I am using Karavan Vs Code extension. I actually don't see an option to create karavan application directly.
How to use Karavan web app directly?
![karavan-create_App](https://github.com/apache/camel-karavan/assets/160023109/321fb985-cace-489f-9662-702be54552e2)

Can you please help me here?

**Reply by MN-Dheeraj** [reply, marked as answer]
Hi @davsclaus 

Also to update on the issue,

It got resolved now. Code is exported now.
Thank you so much.

PS D:\Apache-Camel-Karavan-Integration\projects\POC\Karavan_generated\karavan> jbang "-Dcamel.jbang.version=4.3.0" camel@apache/camel export --fresh  --directory=d:\Apache-Camel-Karavan-Integration\projects\POC\Karavan_generated\karavan\demo_sb --runtime=camel-main --gav=com.demo.camel.karavan:karavan:0.0.1-SNAPSHOT    
Generating fresh run data
Exporting as Camel Main project to: d:\Apache-Camel-Karavan-Integration\projects\POC\Karavan_generated\karavan\demo_sb

But, I have a doubt here. If karavan always 'camel-main', why does it not set that in the command by default.
I actually had run the command the manually on terminal for this. When I click on the 'export' button, this option is not set in the command.
Do you think I missed anything that caused this?

Best Regards,
MN. Dheeraj
